autogen/oai/client.py

```diff
@@ -408,6 +410,40 @@ def cost(self, response: Union[ChatCompletion, Completion]) -> float:
             return (tmp_price1K[0] * n_input_tokens + tmp_price1K[1] * n_output_tokens) / 1000  # type: ignore [no-any-return]
         return tmp_price1K * (n_input_tokens + n_output_tokens) / 1000  # type: ignore [operator]
 
+    def _map_params(self, params: Dict[str, Any]) -> Dict[str, Any]:
+        """Maps parameters that are deprecated"""
+
+        # max_tokens is deprecated and replaced by max_completion_tokens as of 2024.09.12
+        if "max_tokens" in params:
+            params["max_completion_tokens"] = params.pop("max_tokens")
+            logger.warning("OpenAI API: 'max_tokens' parameter is deprecated, converting to 'max_completion_tokens'.")
+
+        if params["model"].startswith("o1"):
```

Comment:
If this is a limitation for o1 preview only, shall we add a check of "preview" in the model name?

Reply:
Max_tokens has been deprecated for all models, with max_completion_tokens working for all models.